### PR TITLE
policy: Add refactor carve out

### PR DESCRIPTION
**Pull request by tcharding**
I have managed to burn out or bore our reviewers/maintainers. Getting two acks is becoming increasingly difficult. I've pestered everyone to the limit that I feel socially comfortable doing so, and as such, am requesting a carve out to the 2-ACK before merge rule.

The primary justification is that I feel we should have a bit more of BDFL and a bit less total consensus if we are to push forwards.

### Example PRs where this change would apply

- https://github.com/rust-bitcoin/rust-bitcoin/pull/1925
- https://github.com/rust-bitcoin/rust-bitcoin/pull/1854
- https://github.com/rust-bitcoin/rust-bitcoin/pull/1862

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK 55be538dac10ceca3da5866a3438fa15dda56327

**Comment by apoelstra** [comment]
Good luck getting other ACKs on this :).

**Comment by yancyribbens** [comment]
In my experience with bitcoin core, if you don't get enough ACKs, it just gets closed instead of merged.  I don't know why rust-bitcoin should be different?

**Comment by apoelstra** [comment]
@yancyribbens the main reason to be different is that Bitcoin Core is much more "feature complete" than this library. As far as its role of being a p2p node enforcing the exact consensus rules, it's more-or-less done. With a few big exceptions, like BIP-324, which are self-contained projects that nonetheless take multiple years to spec and implement.

Another reason is that Bitcoin Core, as developer-starved as it is, actually has more active maintainers and contributors than rust-bitcoin.

Finally, despite both these reasons, Core *does* struggle with a lack of momentum on its non-core functionality. And there's a lot of effort being put into separating out the wallet, GUI, RPC, etc., into separate binaries and eventually even separate repos, so that these other things can move more quickly. It's acknowledged that the Core "most things die on the vine" development model is maybe alright for the core consensus stuff but it's not the right model for less-critical user-facing stuff.

I'd argue that rust-bitcoin doesn't have anything like "core consensus stuff" (maybe the libsecp bindings come close) and that the whole project probably should shift toward a faster-moving model.

**Comment by tcharding** [comment]
Some mentions and further arguments to back up this PR and attempt to get it in:

- @stevenroose suggested that we move too fast and that he wants time to NACK, this fits in with that because it gives two weeks for NACKs 
- @TheBlueMatt mentioned by email that he is too busy so this PR should be ok with him
- @sanket1729 mentioned here on github that he is allocating some time to reviews, 2 weeks should be enough for him to see the things he wants to and ignore the things he does not
- @RCasatta, IIRC said he is bored of all the refactoring
- @Kixunil is on holiday at the moment
- @elichai I can't remember if your position on this whole thing :)

**Review by sanket1729 — APPROVED** [review]
ACK 55be538dac10ceca3da5866a3438fa15dda56327. Same reasons as @apoelstra. And this is only for re-factors that are not adding any new features.

**Comment by sanket1729** [comment]
I guess this PR needs more than 2 ACKS :)

**Comment by apoelstra** [comment]
Yes, I think we should leave this open for 2 weeks to wait for NACKs.

**Comment by tcharding** [comment]
I will email all maintainers to flag this PR, that way any that do not see notifications will not get rugged by this. I sure wish we had a public mailing list so there was public proof that I send the email.

**Comment by yancyribbens** [comment]
For the reasons @apoelstra mentioned, it makes sense for rust-bitcoin to be more permissive than bitcoin.   However I don't think two acks is very uncommon in my experience with other projects and I don't understand why it's so difficult to get two acks here.  Maybe there needs to be more dedicated maintainers?  I'm not sure what the process is to becoming a maintainer.

**Comment by elichai** [comment]
I agree this makes sense for refactors ACK 55be538dac10ceca3da5866a3438fa15dda56327

**Comment by tcharding** [comment]
> I don't understand why it's so difficult to get two acks here.

You could read over the 600 patches that I did in the last 18 months and see if you are still enthusiastic about reviewing refactor work :)

**Comment by yancyribbens** [comment]
> You could read over the 600 patches that I did in the last 18 months and see if you are still enthusiastic about reviewing refactor work :)

I'm fine to help review.  I'm not a maintainer so my acks don't matter however :)

**Review by RCasatta — APPROVED** [review]
ACK 55be538dac10ceca3da5866a3438fa15dda56327

**Review by elichai — APPROVED** [review]

**Comment by stevenroose** [comment]
Yeah I think I can ACK this. I was indeed saying I think a minimim review period would be good. I actually think 2 weeks + 1 ACK is a way saner merge policy than any 2 ACKs (I've seen merge times of less than a day :sweat_smile:)

**Review by stevenroose — APPROVED** [review]

**Comment by yancyribbens** [comment]
Ack https://github.com/rust-bitcoin/rust-bitcoin/commit/55be538dac10ceca3da5866a3438fa15dda56327

**Comment by tcharding** [comment]
That is an ack from every maintainer except @Kixunil.

**Comment by tcharding** [comment]
Can we merge this @apoelstra? I believe that the following three PRs are also good examples of minor fixes that no one cares enough about to review and are harmless to merge.
- https://github.com/rust-bitcoin/rust-bitcoin/pull/1947
- https://github.com/rust-bitcoin/rust-bitcoin/pull/1933
- https://github.com/rust-bitcoin/rust-bitcoin/pull/1914

**Comment by apoelstra** [comment]
Yes, but I also can't think of a way.

And sure, I'll merge this. We are technically 6 hours short of 2 weeks here :P. But we are only waiting on Kix's approval, and I don't think his absense should override six ACKs.

**Comment by tcharding** [comment]
Awesome, onwards and upwards!

**Pull request merged**
